src/App.jsx

```diff
@@ -63,16 +61,12 @@ function App () {
     completePsiturk()
   }
 
-  // Function to add jspsych data on login
+  // Function to capture login data, so we can pass it on to JsPsychExperiment.
   const setLoggedIn = useCallback(
     (loggedIn, studyId, participantId) => {
       if (loggedIn) {
-        jsPsych.data.addProperties({
-          participant_id: participantId,
-          study_id: studyId,
-          start_date: startDate,
-          task_version: taskVersion
-        })
+        setEnvParticipantId(participantId)
+        setEnvStudyId(studyId)
```

Comment:
Another place where I'm trying to keep `jspsych` scoped to the `JsPsychExperiment` component.  Here, I think we can just hang on to the `participantId` and `studyId` and pass them in to `JsPsychExperiment` as props, below.  Then its up to `JsPsychExperiment` to save the prop values to with the `jspsych` data.